src/core/environment_test.go

```diff
@@ -492,6 +493,152 @@ func TestWriteFile(t *testing.T) {
+	"PRIVACY_POLICY_URL=\"https://fedoraproject.org/wiki/Legal:PrivacyPolicy\"\n" +
+	"VARIANT=\"Workstation Edition\"\n" +
+	"VARIANT_ID=workstation"
+}
+
+func TestParseOsReleaseFile(t *testing.T) {
+	tests := []struct {
+		name string
+		osReleaseContent string
+		expect map[string]string
+	} {
+		{
+			name: "TestUbuntuOsReleaseInfo",
+			osReleaseContent: getUbuntuOsReleaseInfo(),
+			expect: map[string]string{
+				"VERSION_ID": "20.04",
+				"VERSION": "20.04.5 LTS (Focal Fossa)",
+				"VERSION_CODENAME": "focal",
+				"NAME": "Ubuntu",
+				"ID": "ubuntu",
+			},
+		},
+		{
+			name: "TestFedoraOsReleaseInfo",
+			osReleaseContent: getFedoraOsReleaseInfo(),
+			expect: map[string]string{
+				"VERSION_ID": "32",
+				"VERSION": "32 (Workstation Edition)",
+				"VERSION_CODENAME": "",
+				"NAME": "Fedora",
+				"ID": "fedora",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			reader := strings.NewReader(tt.osReleaseContent)
+			osRelease, _ := parseOsReleaseFile(reader)
+			assert.Equal(t, tt.expect["VERSION_ID"] , osRelease["VERSION_ID"])
+			assert.Equal(t, tt.expect["VERSION"] , osRelease["VERSION"])
+			assert.Equal(t, tt.expect["VERSION_CODENAME"] , osRelease["VERSION_CODENAME"])
+			assert.Equal(t, tt.expect["NAME"] , osRelease["NAME"])
+			assert.Equal(t, tt.expect["ID"] , osRelease["ID"])
+		})
+	}
+}
+
+func TestMergeHostAndOsReleaseInfo(t *testing.T) {
+	tests := []struct {
+		name string
+		hostRelease *proto.ReleaseInfo
+		osRelease map[string]string
+		expect *proto.ReleaseInfo
+	} {
+		{
+			name: "TestOsReleaseInfoPresent",
+			hostRelease: &proto.ReleaseInfo{
+				VersionId: "20.04",
+				Version:   "5.15.0-1028-aws",
+				Codename:  "linux",
+				Name:      "debian",
+				Id:        "ubuntu",
+			},
+			osRelease: map[string]string{
+				"VERSION_ID": "20.04",
+				"VERSION": "20.04.5 LTS (Focal Fossa)",
+				"VERSION_CODENAME": "focal",
+				"NAME": "Ubuntu",
+				"ID": "ubuntu",
+			},
+			expect: &proto.ReleaseInfo{
+				VersionId: "20.04",
+				Version:   "20.04.5 LTS (Focal Fossa)",
+				Codename:  "focal",
+				Name:      "Ubuntu",
+				Id:        "ubuntu",
+			},
+		},
+		{
+			name: "TestOsReleaseInfoMissing",
+			osRelease: map[string]string{
+				"VERSION_ID": "32",
+				"VERSION": "32 (Workstation Edition)",
+				"VERSION_CODENAME": "",
+				"NAME": "Fedora",
+				"ID": "fedora",
+			},
+			hostRelease: &proto.ReleaseInfo{
+				VersionId: "32",
+				Version:   "Fedora 32 (Workstation Edition)",
+				Codename:  "fedora",
+				Name:      "Fedora",
+				Id:        "Fedora",
+			},
+			expect: &proto.ReleaseInfo{
+				VersionId: "32",
+				Version: "32 (Workstation Edition)",
+				Codename: "fedora",
+				Name: "Fedora",
+				Id: "fedora",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			releaseInfo := mergeHostAndOsReleaseInfo(tt.hostRelease, tt.osRelease)
+			assert.Equal(t, tt.expect.GetVersionId() , releaseInfo.GetVersionId())
```

Comment:
if we use `proto.ReleaseInfo` instead of `*proto.ReleaseInfo`, we can benefit from the `assert.Equal`'s deep comparisons.

For example, given this test:

```go
func TestFoo(t *testing.T) {
	type foo struct {
		name  string
		value int
	}
	expected := foo{"a", 1}
	actual := foo{"b", 2}
	assert.Equal(t, expected, actual)
}
```

we get this output:

```text
--- FAIL: TestFoo (0.00s)
    /path/to/test.go:36: 
        	Error Trace:	/path/to/test.go:36
        	Error:      	Not equal: 
        	            	expected: promise_test.foo{name:"a", value:1}
        	            	actual  : promise_test.foo{name:"b", value:2}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (promise_test.foo) {
        	            	- name: (string) (len=1) "a",
        	            	- value: (int) 1
        	            	+ name: (string) (len=1) "b",
        	            	+ value: (int) 2
        	            	 }
        	Test:       	TestFoo
```

Reply:
It does work on *proto.ReleaseInfo too.

```
--- FAIL: TestMergeHostAndOsReleaseInfo (0.00s)
    --- FAIL: TestMergeHostAndOsReleaseInfo/TestMergeWithOsReleaseInfoFieldMissing (0.00s)
        /Users/a.chawla/workspace/forked-agent-repo/agent/src/core/environment_test.go:653: 
            	Error Trace:	/Users/a.chawla/workspace/forked-agent-repo/agent/src/core/environment_test.go:653
            	Error:      	Not equal: 
            	            	expected: &proto.ReleaseInfo{Codename:"fedora", Id:"fedora", Name:"fedora", VersionId:"32", Version:"32 (Workstation Edition)", XXX_NoUnkeyedLiteral:struct {}{}, XXX_unrecognized:[]uint8(nil), XXX_sizecache:0}
            	            	actual  : &proto.ReleaseInfo{Codename:"fedora", Id:"fedora", Name:"Fedora", VersionId:"32", Version:"32 (Workstation Edition)", XXX_NoUnkeyedLiteral:struct {}{}, XXX_unrecognized:[]uint8(nil), XXX_sizecache:0}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -3,3 +3,3 @@
            	            	  Id: (string) (len=6) "fedora",
            	            	- Name: (string) (len=6) "fedora",
            	            	+ Name: (string) (len=6) "Fedora",
            	            	  VersionId: (string) (len=2) "32",
            	Test:       	TestMergeHostAndOsReleaseInfo/TestMergeWithOsReleaseInfoFieldMissing

Reply:
nice! I learn something new every day about this ecosystem.